Automattic/jetpack
# Move RecordBarMeter to js-packages

#### Changes proposed in this Pull Request:

This PR:

- Moves `RecordMeterBar` component from `plugins/jetpack/_inc/client/components` to `js-packages/components` as discussed in this p2 discussion - pcNPJE-zh-p2#comment-1113.
- Converts `RecordMeterBar` to TypeScript after moving it to js-packages.
- Creates a story for `RecordMeterBar` in TypeScript.
- Adds Unit tests for it written in TypeScript.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

- Run `cd projects/js-packages/storybook && npm run storybook:dev` and confirm that you see `RecordMeterBar` component in Storybook.
- Run unit tests for `js-packages/components` by running `jetpack test`
- Run build, watch commands and unit tests to look for any regression
- Optionally, you may also follow the testing instructions for #23374 to ensure there is no regression.

<img width="1111" alt="Screenshot 2022-04-14 at 12 30 04 PM" src="https://user-images.githubusercontent.com/18226415/163331435-43194a95-0bc7-420b-bcdd-347524dcb530.png">
<img width="1105" alt="Screenshot 2022-04-14 at 12 30 24 PM" src="https://user-images.githubusercontent.com/18226415/163331448-adbe28d7-524c-40e6-a961-d9a8accd5d8d.png">
